ImperatorToCk2/src/main/java/ImperatorToCK2/Output.java

```diff
@@ -939,4 +1041,133 @@ else if (tag.equals("9999")) { //uncolonized province, creates dejure provinces
+
+        ArrayList<String> oldFile = new ArrayList<String>();
+
+        oldFile = Importer.importBasicFile(Directory);
+
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
+        FileOutputStream fileOut= new FileOutputStream(Directory);
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int aqq = 0;
+
+            while (aqq < oldFile.size()) {
+                out.println (oldFile.get(aqq));
+                String key = oldFile.get(aqq).replace(tab,"");
+                key = key.replace("#","");
+                if (key.equals("title = e_TAG")) {
+                    String imperialTag = oldFile.get(aqq).replace("#","");
+                    imperialTag = imperialTag.replace("e_TAG",rank+"_"+title);
+                    out.println (imperialTag);
+                }
+
+                aqq = aqq + 1;
+
+            }
+
+
+        out.flush();
+        fileOut.close();
+
+        return ck2CultureInfo;
+    }
+
+    public static String imperialSuccession(String title, String rank, String Directory) throws IOException //needed to allow TAGs imperial laws
+    {
+
+        String VM = "\\"; 
+        VM = VM.substring(0);
+        char VMq = '"';
+        String tab = "	";
+
+        Directory = Directory + VM + "common" + VM + "laws" + VM + "succession_laws.txt";
+
+        ArrayList<String> oldFile = new ArrayList<String>();
+
+        oldFile = Importer.importBasicFile(Directory);
+
+        String ck2CultureInfo ="a";   // Owner Culture Religeon PopTotal Buildings
+        FileOutputStream fileOut= new FileOutputStream(Directory);
+        PrintWriter out = new PrintWriter(fileOut);
+
+        int aqq = 0;
+
+
+            while (aqq < oldFile.size()) {
+                out.println (oldFile.get(aqq));
+                if (oldFile.get(aqq).contains("e_TAG")) {
+                    String imperialTag = oldFile.get(aqq).replace("e_TAG",rank+"_"+title);
+                    imperialTag = imperialTag.replace("#","");
+                    out.println (imperialTag);
+                }
+
+                aqq = aqq + 1;
+
+            }
+
+
+        out.flush();
+        fileOut.close();
+
+        return ck2CultureInfo;
+    }
+
+    public static void copyRaw(String dir1, String dir2) throws IOException
+    {
+
+        String VM = "\\";
+        VM = VM.substring(0);
+
+        FileInputStream fileIn= new FileInputStream(dir1);
+        FileOutputStream fileOut= new FileOutputStream(dir2);
+
+        boolean endOrNot = true;
+        int aqq = 0;
+
+        try {
+            while (endOrNot = true && aqq != -1){
+                if (aqq != -1) {
+                    aqq = fileIn.read();
+                    fileOut.write(aqq);
+                }
+
+            }
+        }catch (java.util.NoSuchElementException exception){
+            endOrNot = false;
+            fileIn.close();
+            fileOut.close();
+
+        }   
+
+    }
```

Comment:
Instead of having this function, use one of the methods here: https://www.baeldung.com/java-copy-file
I'd recommend #3

Reply:
The function I wrote works fine, I think that it's more trouble then it's worth to change it, especially with the other work that needs to be done before release

Reply:
I'm going to be hard-nosed about this. 
1. Overly-complex code tends to infect a project, as it makes it easier for more overly-complex code to slip in.
2. It'll take you less time to do the update than it did to reply. Here's the relevant line: ```Files.copy(dir1, dir2, StandardCopyOption.REPLACE_EXISTING)```